cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -344,13 +342,40 @@ def _generate_purl_for_package(package: _ResolvedPackage, project: Project) -> s
     :param project: the project object to resolve the configured registry url and file paths
         for file dependencies.
     """
-    # registry url can be accessed in project.yarnrc
-    # paths for file dependencies are relative to project.source_dir
+    qualifiers = dict()
+    if package.checksum is not None:
+        qualifiers["checksum"] = package.checksum
```

Comment:
I believe we can keep the checksums exclusively for HttpsLocator packages. On the other hand, we should report whichever packages didn't have checksums verified, but that's another story (and that will be done via a property, not purls). Right @taylormadore ^^'?